src/api/Volkszaehler.cpp

```diff
@@ -316,10 +316,10 @@ int vz::api::curl_custom_debug_callback(CURL *curl, curl_infotype type, char *da
 
 	case CURLINFO_SSL_DATA_OUT:
 	case CURLINFO_DATA_OUT:
-		//data[size] = 0;
+		std::string buffer(data, size);
 		print((log_level_t)(log_debug + 5), "CURL: Sent %lu bytes.. ", ch->name(),
 			  (unsigned long)size);
-		print((log_level_t)(log_debug + 5), "CURL: Sent '%s' bytes", ch->name(), data);
+		print((log_level_t)(log_debug + 5), "CURL: Sent '%s' bytes", ch->name(), buffer.c_str());
```

Comment:
thx! would be good to do the same in line 305 as well. We should not modify the data.
And to compensate for the additional malloc/new inside std::string we could e.g. make just one print:
..."CURL: Sent %lu bytes: '%s'", ch->name(), (unsigned long)size, buffer.c_str())